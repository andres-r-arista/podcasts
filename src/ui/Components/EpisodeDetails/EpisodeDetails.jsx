import React from "react";

import PropTypes from "prop-types";

export const EpisodeDetails = ({ title, description, episodeUrl }) => {
  return (
    <div className="border m-4 shadow-md max-w-full px-4 py-4">
      <h1 className="font-bold text-lg py-2" data-testid="episode-details-title">{title}</h1>
      <h2
        className="italic text-sm pb-4 border-b"
        data-testid="episode-details-description"
        dangerouslySetInnerHTML={{ __html: description }}
      />
      <audio controls className="w-full mt-4" >
        <source src={episodeUrl} type="audio/mpeg" data-testid="episode-details-audio"/>
        Your browser does not support the audio element.
      </audio>
    </div>
  );
};

EpisodeDetails.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  episodeUrl: PropTypes.string.isRequired,
};
