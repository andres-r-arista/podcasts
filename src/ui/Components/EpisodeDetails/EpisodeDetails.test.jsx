import { render, screen } from "@testing-library/react";

import { EpisodeDetails } from "./EpisodeDetails";

const title = "Title";
const episodeUrl = "http://www.google.com";
const description = "Description";

describe("Tests in <EpisodeDetails />", () => {
  test("must match the snapshot", () => {
    const { container } = render(
      <EpisodeDetails
        title={title}
        description={description}
        episodeUrl={episodeUrl}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("must display the correct title", () => {
    render(
      <EpisodeDetails
        title={title}
        description={description}
        episodeUrl={episodeUrl}
      />
    );
    const screenTitle = screen.getByTestId("episode-details-title");
    expect(screenTitle).toHaveTextContent("Title");
  });

  test("must display the correct description", () => {
    render(
      <EpisodeDetails
        title={title}
        description={description}
        episodeUrl={episodeUrl}
      />
    );
    const screenTitle = screen.getByTestId("episode-details-description");
    expect(screenTitle).toHaveTextContent("Description");
  });

  test("must use the correct audio url", () => {
    render(
      <EpisodeDetails
        title={title}
        description={description}
        episodeUrl={episodeUrl}
      />
    );
    const screenAudio = screen.getByTestId("episode-details-audio");
    expect(screenAudio.src).toContain("http://www.google.com");
  });
});
