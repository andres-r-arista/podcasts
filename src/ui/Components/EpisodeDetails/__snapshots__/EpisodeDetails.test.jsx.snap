// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <EpisodeDetails /> must match the snapshot 1`] = `
<div>
  <div
    class="border m-4 shadow-md max-w-full px-4 py-4"
  >
    <h1
      class="font-bold text-lg py-2"
      data-testid="episode-details-title"
    >
      Title
    </h1>
    <h2
      class="italic text-sm pb-4 border-b"
      data-testid="episode-details-description"
    >
      Description
    </h2>
    <audio
      class="w-full mt-4"
      controls=""
    >
      <source
        data-testid="episode-details-audio"
        src="http://www.google.com"
        type="audio/mpeg"
      />
      Your browser does not support the audio element.
    </audio>
  </div>
</div>
`;
